core/src/main/scala/com/metamx/tranquility/druid/DruidBeamMaker.scala

```diff
@@ -243,6 +243,7 @@ object DruidBeamMaker
     val tsUtc = new DateTime(ts.millis, ISOChronology.getInstanceUTC)
 
     val cycleBucket = segmentGranularity match {
+      case Granularity.SECOND => (tsUtc.minuteOfHour().get * 60 + tsUtc.secondOfMinute().get) % 900 // 900 buckets
```

Comment:
Without increasing the number of characters allowed for this portion of the base firehouse ID, 999 buckets is the most we can create. I chose 900 buckets because that is an even distribution of seconds over an hour (3600 / 4) while remaining under the 3 digit cap. What I wrote will sequentially fill 900 buckets and then reset after the 900th bucket every 15 mins.